### PR TITLE
Migrate `kedro-airflow` to static metadata

### PR DESCRIPTION
**Pull request by astrojuanlu**
## Description
See https://github.com/kedro-org/kedro/issues/2334.

Also, I saw `kedro-datasets` was missing explicit build-system, so I added it (I think pip has a fallback for cases like this, but better not to rely on it).

## Development notes
I merged `requirements.txt`, `setup.cfg` and `setup.py` into a single `pyproject.toml`, yay!

Differences in `RECORD` (what goes into the wheel):

```diff
--- dist.old/kedro_airflow-0.5.1.dist-info/RECORD       2023-04-17 13:49:26
+++ dist.new/kedro_airflow-0.5.1.dist-info/RECORD       2023-04-17 13:54:00
@@ -1,7 +1,7 @@
 kedro_airflow/__init__.py,sha256=rPTM7Yfk6BHZqBvDv8iFF6DXQ0kNAw24pIHIFdzOZ4w,78
 kedro_airflow/airflow_dag_template.j2,sha256=1hF1L8DF1jUC1QnR6WIzIAkQBHr70FUebVeXQX6pW6s,2565
 kedro_airflow/plugin.py,sha256=Sh7w91mjhstzv5L5g-dTKqeBGTDx8_gMk8pGY7MRi-8,3018
-kedro_airflow-0.5.1.dist-info/METADATA,sha256=LFllLrI94LBNdMXtbAYq1Rdh9vLY40Iy3obsjdEfZ3U,3695
+kedro_airflow-0.5.1.dist-info/METADATA,sha256=2UDgUC5yMq6Qqk1VbGGzXN3XP2JqHKQEbMHjraUIRHc,3882
 kedro_airflow-0.5.1.dist-info/WHEEL,sha256=pkctZYzUS4AYVn6dJ-7367OJZivF2e8RA9b_ZBjif18,92
 kedro_airflow-0.5.1.dist-info/entry_points.txt,sha256=AAmoJsZRgb4tG0WeptLbvvjjmbEDF_1JcsxQVKypT20,65
 kedro_airflow-0.5.1.dist-info/top_level.txt,sha256=qnSLh-V5c3yVXYcKflrXSLyrWkeGZKEMSgLz3Nq2_2c,14
```

Differences in `METADATA`:

```diff
--- dist.old/kedro_airflow-0.5.1.dist-info/METADATA     2023-04-17 13:49:26
+++ dist.new/kedro_airflow-0.5.1.dist-info/METADATA     2023-04-17 13:48:06
@@ -2,10 +2,12 @@
 Name: kedro-airflow
 Version: 0.5.1
 Summary: Kedro-Airflow makes it easy to deploy Kedro projects to Airflow
-Home-page: https://github.com/kedro-org/kedro-plugins/tree/main/kedro-airflow
 Author: Kedro
 License: Apache Software License (Apache 2.0)
-Requires-Python: >=3.7, <3.11
+Project-URL: Source, https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets
+Project-URL: Documentation, https://github.com/kedro-org/kedro-plugins/blob/main/kedro-airflow/README.md
+Project-URL: Tracker, https://github.com/kedro-org/kedro-plugins/issues
+Requires-Python: <3.11,>=3.7
 Description-Content-Type: text/markdown
 Requires-Dist: kedro (>=0.17.5)
 Requires-Dist: python-slugify (>=4.0)
```

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Comment by astrojuanlu** [comment]
Ooops, CI is using `requirements.txt`. Will have a look.

File: kedro-airflow/pyproject.toml [hunk 1]
```diff
@@ -1,8 +1,48 @@
...
+license = {text = "Apache Software License (Apache 2.0)"}
+dependencies = [
+    "kedro>=0.17.5",
+    "python-slugify>=4.0",
+    "semver~=2.10",  # Needs to be at least 2.10.0 to get VersionInfo.match
+]
+dynamic = ["readme", "version"]
+
+[project.urls]
+Source = "https://github.com/kedro-org/kedro-plugins/tree/main/kedro-datasets"
```
Comment (merelcht) [inline kedro-airflow/pyproject.toml:21]:
```suggestion
Source = "https://github.com/kedro-org/kedro-plugins/tree/main/kedro-airflow"
```

**Review by merelcht — APPROVED** [review]
Might be good to add this to the release notes for our own records.

File: kedro-airflow/MANIFEST.in
Comment (AntonyMilneQB) [inline kedro-airflow/MANIFEST.in:1]: I presume there's no new `[tool.setuptools]` configuration that could replace this?

**Review by AntonyMilneQB — APPROVED** [review]
This is heroic work, thank you for doing it!

I have the same thing to double check about the entrypoints as in https://github.com/kedro-org/kedro-plugins/pull/174#pullrequestreview-1389698086, but LGTM 👍

File: kedro-airflow/MANIFEST.in
Comment (AntonyMilneQB) [reply kedro-airflow/MANIFEST.in:1]: I guess not given that you introduced a MANIFEST.in in https://github.com/kedro-org/kedro-plugins/pull/173...

Comment (astrojuanlu) [reply kedro-airflow/MANIFEST.in:1]: `MANIFEST.in` controls what's in the source distribution https://packaging.python.org/en/latest/guides/using-manifest-in/, and then setuptools uses that by virtue of `include-package-data = True` https://setuptools.pypa.io/en/latest/userguide/datafiles.html

However, the `packaging.python.org` docs are a bit outdated and I'm not sure how applicable they are in a post-PEP 621 world... The official Python docs are not much better https://docs.python.org/3/distutils/sourcedist.html?highlight=manifest

To be honest I'm not 100 % sure whether `MANIFEST.in` can be ditched completely in favor of setuptools `package_data`, I'd need to investigate a bit and unfortunately I don't have much time for that at the moment (these simple PRs on the other hand took mere minutes to put together). It's an excellent question but I'd prefer to explore it in a separate PR.

**Pull request merged**